### ISSUE TITLE
Composer bisect command

### ISSUE THREAD

**Issue by Seldaek** [composer/composer#11119]
### Discussed in https://github.com/composer/composer/discussions/11072

<div type='discussions-op-text'>

<sup>Originally posted by **mad-briller** September 21, 2022</sup>
This is just a random idea i've had while upgrading projects and having a hard time narrowing down issues, not sure if issues or discussions is the right place to put this.

Even with all the will in the world to respect semver, packages inevitability break backwards compatibility in ways they did not forsee, everyone is human afterall.

Chasing the version that introduced the issue can be quite hard currently, as you have to load up packagist and look at released versions and install them one by one to see if they are the issue.
This reduces the likelihood that a developer will report the issue to the package maintainers, and also makes the developer less likely to actually update the package, as its safer to stay on the "last working version" and put an explicit version number in `composer.json`.

When a incorrect change is introduced in git, `git bisect` makes it super easy to chase the commit that introduced the issue. It would be great if composer had a `composer bisect <package>` that worked similarly. 
This would make narrowing down which version introduced a bc break much easier. 

Thanks for your time.

</div>

**Comment by greg0ire** [comment]
Here is how I work around this for now: https://dev.to/greg0ire/bisecting-vendors-12kd

**Comment by stof** [comment]
@greg0ire in your case, you are still bisecting an exact git commit. As far as I can tell, the `composer bisect` feature would be bisecting releases (when you update less frequently and so your `composer update` jumped by many versions at once). So that's not the same use case IMO.

**Comment by greg0ire** [comment]
You're right `composer bisect` would go less far indeed, I did not realize that. But it would be a good first step, easier to use, especially in the case of splits such as `symfony/*`.

**Comment by sebastianbergmann** [comment]
I seem to have the same use case as @greg0ire: https://github.com/sebastianbergmann/phpunit/issues/5658#issuecomment-1890862518
